reflectionserializer/struct_field_cache.go

```diff
@@ -0,0 +1,116 @@
+package reflectionserializer
+
+import (
+	"fmt"
+	"go/types"
+	"reflect"
+	"strconv"
+	"sync"
+	"unicode"
+)
+
+type fieldCache struct {
+	lock             sync.Mutex
+	structFieldCache map[reflect.Type][]FieldMetadata
+}
+
+// newFieldCache creates and returns new fieldCache.
+func newFieldCache() *fieldCache {
+	return &fieldCache{
+		structFieldCache: make(map[reflect.Type][]FieldMetadata),
+	}
+}
+
+// FieldMetadata contains information necessary to serialize and deserialize a struct field
+type FieldMetadata struct {
```

Comment:
Perhaps makes sense to also add the field name so it can be used in the context where errors needs to be returned?